tsdb/chunkenc/chunk_test.go

```diff
@@ -217,3 +217,46 @@ func benchmarkAppender(b *testing.B, newChunk func() Chunk) {
 
 	fmt.Println("num", b.N, "created chunks", len(chunks))
 }
+
+func TestXor(t *testing.T) {
+	timestamp := int64(1234123324)
+	v := 83665193758603.123
+	num := 10
+	var samples []pair
+	for i := 0; i < num; i++ {
+		timestamp += int64(1000)
+		v += float64(100)
+		samples = append(samples, pair{t: timestamp, v: v})
+	}
+
+	chunk := NewXORChunk()
+	a, err := chunk.Appender()
+	if err != nil {
+		t.Fatalf("get appender: %s", err)
+	}
+	for _, p := range samples {
+		a.Append(p.t, p.v)
+	}
+
+	res := make([]pair, 0)
+	var it Iterator
+	it = chunk.Iterator(it)
+	for it.Next() {
+		t, v := it.At()
+		res = append(res, pair{t: t, v: v})
+	}
+	if it.Err() != io.EOF {
+		testutil.Ok(t, it.Err())
+	}
+
+	for i := 0; i < len(samples); i++ {
```

Comment:
```suggestion
	for i, sample := range samples {
```

Reply:
r := res[i]

There is also a "res" array that needs to be looped. If there is no "I" index, it will be strange to loop the "res" array

Reply:
Then can you add `testutil.Equals(t, len(res), len(samples))` ?

Reply:
![image](https://user-images.githubusercontent.com/9583245/75448207-b935ad00-59a5-11ea-98a8-5d0a24dc73a2.png)
done,thanks